### PR TITLE
Fix root/meta for ArraySerializer

### PR DESCRIPTION
**Pull request by ryansch**

### CODE REVIEW

**Comment by ryansch** [comment]
I've rebased this on master and squashed in a commit to make both calls to `#assert_equal` use the correct argument order.

File: lib/active_model/serializer/array_serializer.rb [hunk 1]
```diff
@@ -21,11 +22,11 @@ def initialize(objects, options = {})
       end
 
       def json_key
-        @objects.first.json_key if @objects.first
-      end
-
-      def root=(root)
-        @objects.first.root = root if @objects.first
+        if root == true && @objects.first
+          @objects.first.class.root_name
```
Comment (joaomdmoura) [inline lib/active_model/serializer/array_serializer.rb]: I'm not sure about using the ```root_name``` here. ArraySerializer is pretty abstract, take a lott at this piece of code for example:

```ruby
@objects = objects.map do |object|
  serializer_class = options.fetch(
    :serializer,
    ActiveModel::Serializer.serializer_for(object)
  )
  serializer_class.new(object, options.except(:serializer))
end
```

By checking this code we can presume that the objects might not be instances from the same class. Wrap them by the class name from its first object doesn't would be right.

Comment (ryansch) [inline lib/active_model/serializer/array_serializer.rb]: The problem here is the jsonapi adapter.  It sets `serializer.root = true` in its initializer.  I'm not sure how to deal with heterogenous arrays.

File: lib/active_model/serializer/adapter/json.rb [hunk 2]
```diff
@@ -4,6 +4,11 @@ module ActiveModel
   class Serializer
     class Adapter
       class Json < Adapter
+        def initialize(serializer, options = {})
+          super
+          serializer.root = true if @options[:root]
```
Comment (bf4) [inline lib/active_model/serializer/adapter/json.rb]: this logic doesn't make sense to me.  why are we only checking if the value of options[:root] is truthy?

File: lib/active_model/serializer/array_serializer.rb [hunk 3]
```diff
@@ -4,9 +4,10 @@ class ArraySerializer
       include Enumerable
       delegate :each, to: :@objects
 
-      attr_reader :meta, :meta_key
+      attr_accessor :meta, :meta_key, :root
```
Comment (bf4) [inline lib/active_model/serializer/array_serializer.rb]: what's the read to change the reader to an accessor?

File: lib/active_model/serializer/array_serializer.rb (hunk 1 above)
Comment (joaomdmoura) [inline lib/active_model/serializer/array_serializer.rb]: We could define a default ```data``` root by default is there isn't one defined. But I'm not convinced yet about the root implementation on ```ArraySerializer``` :stuck_out_tongue_closed_eyes:

**Comment by ryansch** [comment]
I'm going to retire this PR.  We ended up ditching both jsonapi and AMS 0.10.x in favor of 0.8.3 for now.  More of a 'just works' approach.

**Comment by joaomdmoura** [comment]
:cry:

**Comment by bf4** [comment]
@joaomdmoura This is fixed in #936, no?

**Comment by joaomdmoura** [comment]
@bf4 I haven't test, so I'm not sure about it. But supposedly yes.

**Pull request closed without merging**